packages/react-components/src/components/table-output-component.tsx

```diff
@@ -252,12 +272,16 @@ export class TableOutputComponent extends AbstractOutputComponent<TableOutputPro
     }
 
     private onModelUpdated = async () => {
-        if (this.pendingIndex && this.gridApi) {
-            if (this.gridApi.getSelectedNodes().length === 0) {
-                this.gridApi.getDisplayedRowAtIndex(this.pendingIndex).setSelected(true);
+        this.pendingIndices.forEach(index => {
+            if (this.gridApi) {
+                const node = this.gridApi.getDisplayedRowAtIndex(index);
+                if (node.id) {
+                    node.setSelected(true);
+                    this.gridApi.ensureIndexVisible(index);
```

Comment:
You only need to ensure that the endIndex is visible. This should be the last (most recent) selected time, e.g with Shift+right it is the last selected row, with Shift-left it is the first selected row.